test/test_outlier_detection.py

```diff
@@ -0,0 +1,495 @@
+- Different numbers of providers (2, 3, 5, 7)
+- Different magnitudes of outliers (2x, 3x, 5x, 10x)
+- Edge cases (zeros, NaN, identical values, small values)
+- Configuration options
+"""
+
+import numpy as np
+import pandas as pd
+
+from traveltime_drive_time_comparisons.common import (
+    Fields,
+    GOOGLE_API,
+    TRAVELTIME_API,
+    TOMTOM_API,
+    HERE_API,
+    OSRM_API,
+    MAPBOX_API,
+    VALHALLA_API,
+)
+from traveltime_drive_time_comparisons.outlier_detection import (
+    OutlierConfig,
+    detect_outliers,
+    filter_outliers,
+)
+
+
+class TestOutlierDetectionBasics:
+    """Test basic outlier detection functionality."""
+
+    def test_detect_obvious_high_outlier_with_7_providers(self):
+        """5x outlier should be detected with 7 providers."""
+        data = pd.DataFrame(
+            {
+                Fields.TRAVEL_TIME[GOOGLE_API]: [1000, 1000, 1000, 1000],
+                Fields.TRAVEL_TIME[TRAVELTIME_API]: [1000, 1000, 1000, 1000],
+                Fields.TRAVEL_TIME[TOMTOM_API]: [1000, 5000, 1000, 1000],
+                Fields.TRAVEL_TIME[HERE_API]: [1000, 1000, 1000, 1000],
+                Fields.TRAVEL_TIME[OSRM_API]: [1000, 1000, 1000, 1000],
+                Fields.TRAVEL_TIME[MAPBOX_API]: [1000, 1000, 1000, 1000],
+                Fields.TRAVEL_TIME[VALHALLA_API]: [1000, 1000, 1000, 1000],
+            }
+        )
+
+        config = OutlierConfig(ratio_threshold=3.0)
+        outliers = detect_outliers(data, Fields.TRAVEL_TIME, config)
+
+        assert len(outliers) == 1
+        assert outliers.iloc[0]["row_index"] == 1
+        assert outliers.iloc[0]["outlier_provider"] == "TomTom"
+        assert outliers.iloc[0]["outlier_value"] == 5000
+        assert outliers.iloc[0]["median_value"] == 1000
+        assert outliers.iloc[0]["ratio"] == 5.0
+
+    def test_detect_obvious_low_outlier_with_7_providers(self):
+        """10x low outlier should be detected with 7 providers."""
+        data = pd.DataFrame(
+            {
+                Fields.TRAVEL_TIME[GOOGLE_API]: [500, 500, 500, 50],
+                Fields.TRAVEL_TIME[TRAVELTIME_API]: [500, 500, 500, 500],
+                Fields.TRAVEL_TIME[TOMTOM_API]: [500, 500, 500, 500],
+                Fields.TRAVEL_TIME[HERE_API]: [500, 500, 500, 500],
+                Fields.TRAVEL_TIME[OSRM_API]: [500, 500, 500, 500],
+                Fields.TRAVEL_TIME[MAPBOX_API]: [500, 500, 500, 500],
+                Fields.TRAVEL_TIME[VALHALLA_API]: [500, 500, 500, 500],
+            }
+        )
+
+        config = OutlierConfig(ratio_threshold=3.0)
+        outliers = detect_outliers(data, Fields.TRAVEL_TIME, config)
+
+        assert len(outliers) == 1
+        assert outliers.iloc[0]["row_index"] == 3
+        assert outliers.iloc[0]["outlier_value"] == 50
+        assert outliers.iloc[0]["ratio"] == 10.0
+
+    def test_no_outlier_with_natural_variance(self):
+        """Natural variance (±10%) should not be flagged."""
```

Comment:
We are doing an assumption of what the natural variance is. That's a weakness of the current approach, compared to the IQR method, which is automatically detecting the variance in the dataset. 
It may be that you found IQR to be super sensitive because in most of the test data the variance is unnaturally low. E.g. testing such data with IQR:
data = [1000, 1000, 1000, 1000, 1010]
the 1010 is flagged as an outlier, but it's not the problem with the method, but with the test data.

Having said that, I'm ok to proceed with current approach, but if it doesn't behave as expected in the wild, this may be the reason.